shared/rocroller/test/catch/RotatingBufferTest.cpp

```diff
@@ -26,33 +25,46 @@
  *******************************************************************************/
 
 #include "client/RotatingBuffer.hpp"
+#include <algorithm>
 #include <catch2/catch_template_test_macros.hpp>
 #include <catch2/catch_test_macros.hpp>
 #include <catch2/matchers/catch_matchers_string.hpp>
 #include <cstddef>
+#include <hip/hip_runtime.h>
 
 using namespace rocRoller;
 
-TEST_CASE("Disabled rotation returns base pointer", "[RotatingBuffer]")
+template <typename T>
+static std::vector<T> copyDeviceSpanToHost(std::span<T> dSpan)
+{
+    std::vector<T> h(dSpan.size());
+    // hipMemcpy expects void*; cast the device pointer accordingly
+    HIP_CHECK(hipMemcpy(h.data(), dSpan.data(), sizeof(T) * dSpan.size(), hipMemcpyDeviceToHost));
+    return h;
+}
+
+TEST_CASE("RotatingBuffer: Disabled rotation returns base pointer", "[RotatingBuffer]")
 {
     std::vector<float>    hostData(16, 1.0f);
-    RotatingBuffer<float> buf(hostData, 0); // rotation disabled
+    RotatingBuffer<float> buf(hostData, 0);
 
     auto span1 = buf.next();
     auto span2 = buf.next();
 
-    REQUIRE(span1.data() == span2.data()); // always base pointer
+    REQUIRE(span1.data() == span2.data());
     REQUIRE(span1.size() == hostData.size());
-    for(auto v : span1)
+
+    auto h1 = copyDeviceSpanToHost(span1);
+    for(auto v : h1)
     {
         REQUIRE(v == 1.0f);
     }
```

Comment:
Can we use our custom catch2 matcher instead?
```suggestion
    REQUIRE_THAT(span1.data(), HasDeviceVectorEqualTo(hostData));
```

Reply:
Sure, I replaced the manual host copy + value loop checks with our `HasDeviceVectorEqualTo` custom matcher.